metis-harvesting/src/main/java/eu/europeana/metis/harvesting/ReportingIteration.java

```diff
@@ -11,19 +11,13 @@
 @FunctionalInterface
 public interface ReportingIteration<T> {
 
-  /**
-   * The result of an iteration, indicating whether to proceed.
-   */
-  enum IterationResult {TERMINATE, CONTINUE}
-
   /**
    * Perform one iteration for the given data.
    *
    * @param data The data to process.
    * @return Whether to continue processing.
-   * @throws IOException in case there was a harvesting related issue. This will cause the remaining
-   *                     records not to be processed (as if {@link IterationResult#TERMINATE} was
-   *                     passed).
+   * @throws IOException in case there was a harvesting-related issue. This will cause the remaining
+   *                     records not to be processed.
    */
-  IterationResult process(T data) throws IOException;
+  boolean process(T data) throws IOException;
```

Comment:
Is a personal taste, to have a more descriptive response of what the function returns`Terminate` or `Continue`. In terms of performance I think a boolean is simplifies it. Just remember to add the proper documentation of what means `false` and `true` of this function in the `javadoc`.